micro-rdk/src/esp32/nvs_storage.rs

```diff
@@ -126,6 +126,40 @@ impl NVSStorage {
     }
 }
 
+impl StorageDiagnostic for NVSStorage {
+    type Error = NVSStorageError;
+    fn log_space_diagnostic(&self) -> Result<(), Self::Error> {
+        let part_name = CString::new("nvs")
```

Comment:
Does this only work if the name of the partition is `nvs`? I don't think esp-idf requires that it be called that, and I think the NVS library will use all partitions of type `data` and subtype `nvs`.

Short of dealing with that complexity, could you dig out the name from `self.nvs` instead? I think that way it would honor the `partition_name` `str` passed to `NVSStorage::new`.